lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -41,6 +41,38 @@ def object
           end
         end
 
+        # Parse a Hash or ActionController::Parameters representing a JSON API document
+        # into an ActiveRecord-ready hash.
+        # NOTE(beauby): Currently this does not handle relationships with modified keys.
+        #
+        # @param [Hash|ActionController::Parameters] document
+        # @return [Hash] ActiveRecord-ready hash
+        #
+        def self.parse(document)
+          hash = {}
+
+          hash[:id] = document['data']['id'] if document['data']['id']
```

Comment:
Sure. C.f. the description of the PR, there is currently no error handling, but it is planned.